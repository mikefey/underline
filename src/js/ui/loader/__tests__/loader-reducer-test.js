import test from 'ava';
import {
  SHOW_LOADER,
  HIDE_LOADER,
} from './../loader-constants';
import loaderReducer from './../loader-reducer.js';

test('default state', (t) => {
  t.snapshot(loaderReducer(undefined, {}));
});

test('handles SHOW_LOADER', (t) => {
  t.snapshot(loaderReducer(undefined, {
    type: SHOW_LOADER,
    message: 'Loading',
  }));
});

test('handles HIDE_LOADER', (t) => {
  t.snapshot(loaderReducer(undefined, { type: HIDE_LOADER }));
});
