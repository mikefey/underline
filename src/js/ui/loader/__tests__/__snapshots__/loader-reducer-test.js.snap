// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default state 1`] = `
Object {
  "message": "",
  "show": false,
}
`;

exports[`handles HIDE_LOADER 1`] = `
Object {
  "message": "",
  "show": false,
}
`;

exports[`handles SHOW_LOADER 1`] = `
Object {
  "message": "Loading",
  "show": true,
}
`;
